### ISSUE TITLE
No formatter installed for "ansible" file type.

### ISSUE THREAD

**Issue by jamesharris-garmin** [ansible/vscode-ansible#1110]
### Summary

TLDR: When I attempt to format document with `Shift` _ `Alt` + `F` visual studio code prompts me to install a formatter even though I already have the redhat yaml extension installed.

Additional notes:

Redhat yaml extension is:
redhat.vscode-yaml version (v1.14.0)

VSCODE's install ansible formatter search gives:

![image](https://github.com/ansible/vscode-ansible/assets/52711649/593c6c94-8687-4e55-8040-d8c40b77475a)

None of these extensions work as is.

### Extension version

v2.12.143

### VS Code version

Version: 1.87.0

### Ansible Version

```shell
$ ansible --version
ansible [core 2.15.9]
  config file = None
  configured module search path = ['/home/harrisja/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/harrisja/.local/pipx/venvs/venvs/ansible/lib/python3.10/site-packages/ansible
  ansible collection location = /home/harrisja/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/harrisja/.local/bin/ansible
  python version = 3.10.13 (main, Aug 25 2023, 13:20:03) [GCC 9.4.0] (/home/harrisja/.local/pipx/venvs/venvs/ansible/bin/python)
  jinja version = 3.1.3
  libyaml = True
```

### OS / Environment

Windows 11, remote to Ubuntu 22.04 (Jammy), 
ansible-lint 24.2.0

**Comment by priyamsahoo** [comment]
This is a compatible issue with the YAML extension. We have default formatting set up in our extension, and we do not want to rely on the YAML extension to do it.

Having said that, you can always set up a default extension for `ansible` language and point it to the `YAML extension` or `prettier.` `prettier` seems to work well with the `ansible` language. If the YAML extension is not formatting it, you might have to raise an issue with their extension to support formatting for `ansible` language.

Here's how you can set up the formator. Add these to your settings configuration:
```json
"[ansible]": {
    "editor.autoIndent": "advanced",
    "editor.defaultFormatter": "redhat.vscode-yaml",
    "editor.formatOnSave": true
  }
```

**Comment by jamesharris-garmin** [comment]
That sounds reasonable. I'll raise a ticket for the vscode-yaml extension to cover the ansible language.

**Comment by lqhuang** [comment]
@priyamsahoo This doesn't work.

```
"[ansible]": {
    "editor.autoIndent": "advanced",
    "editor.defaultFormatter": "redhat.vscode-yaml",
    "editor.formatOnSave": true
}
```

See https://github.com/redhat-developer/vscode-yaml/issues/812

Since `redhat.vscode-yaml` doesn't register Ansible. Yeah, that's crazy, they're both in the same `redhat` namespace.
